README.md

```diff
@@ -22,10 +22,125 @@ The lack of a standardized, cryptographic verification process for exchange solv
+     * @param timestamp MUST be the timestamp at which the CEX took the snapshot of its assets and liabilites
+     * @param mstRoot MUST be the Merkle sum tree root of the CEX's liabilities
+     * @param assets MUST be the list of assets owned by the CEX
+     */
+    event SolvencyProofSubmitted(
+        uint256 indexed timestamp,
+        uint256 mstRoot,
+        Asset[] assets
+    );
+
+    /**
+     * @dev Submit an optimistic proof of address ownership for a CEX. The proof is subject to an off-chain verification as it's not feasible to verify the signatures of non-EVM chains in an Ethereum smart contract.
+     * @param _addressOwnershipProofs MUST be the list of address ownership proofs
+     * MUST revert if `_addressOwnershipProofs` array is empty.
+     * MUST revert if any `AddressOwnershipProof` has an empty `cexAddress`, `chain`, `signature`, or `message`.
+     * MUST revert if the `cexAddress` has already been verified.
+     * MUST revert if called by anyone other than the contract owner.
+     * MUST emit `AddressOwnershipProofSubmitted` event upon successful submission.
+     */
+    function submitProofOfAddressOwnership(
+        AddressOwnershipProof[] memory _addressOwnershipProofs
+    ) external;
+
+    /**
+     * @dev Submit proof of solvency of a CEX
+     * @param mstRoot MUST be the Merkle sum tree root of the CEX's liabilities
+     * @param assets MUST be the list of assets owned by the CEX
+     * @param proof MUST be the ZK proof generated from the specified proof of solvency circuit
+     * @param timestamp MUST be the timestamp at which the CEX took the snapshot of its assets and liabilites
+     * In case of successful proof submission `mstRoot` MUST be stored in the contract state against the given `timestamp` because it MAY be used by `verifyProofOfInclusion` function to verify the proof of user inclusion into the Merkle sum tree of the CEX's liabilities. `verifyProofOfInclusion` MUST be able to look up the stored `mstRoot` by a timestamp.
+     * MUST revert if `mstRoot` is zero.
+     * MUST revert if `timestamp` is zero.
+     * MUST revert if `assets` array is empty.
+     * MUST revert if any `Asset` has an empty `chain` or `assetName`.
+     * MUST revert if any `Asset` has an `amount` of zero.
+     * MUST revert if the ZK proof verification fails.
+     * MUST revert if called by anyone other than the contract owner.
+     * MUST emit `AddressOwnershipProofSubmitted` event upon successful submission.
+     */
+    function submitProofOfSolvency(
+        uint256 mstRoot,
+        Asset[] memory assets,
+        bytes memory proof,
+        uint256 timestamp
+    ) external;
+
+    /**
+     * @dev Verify a ZK proof of user inclusion in the Merkle sum tree of the CEX's liabilities
+     * @param proof MUST be a ZK proof of inclusion generated from the specified proof of inclusion circuit
+     * @param timestamp MUST be the timestamp at which the CEX took the snapshot of its assets and liabilites
```

Comment:
I would see "timestamp MUST be **a** timestamp at which the CEX took the snapshot of its assets and liabilities" 

Since there may different timestamps